File: XMLParsingTest/app/src/main/java/com/dstrube/xmlparsingtest/view/SingleMenuItemActivity.java
package com.dstrube.xmlparsingtest.view;

import com.dstrube.xmlparsingtest.R;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class SingleMenuItemActivity extends Activity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		setContentView(R.layout.single_list_item);

		// getting intent data
		Intent in = getIntent();

		// Get XML values from previous intent
		// XML node keys
		String KEY_NAME = "name";
		String name = in.getStringExtra(KEY_NAME);
		String KEY_COST = "cost";
		String cost = in.getStringExtra(KEY_COST);
	String KEY_DESC = "description";
	String description = in.getStringExtra(KEY_DESC);

		// Displaying all values on the screen
		TextView lblName = (TextView) findViewById(R.id.name_label);
		TextView lblCost = (TextView) findViewById(R.id.cost_label);
		TextView lblDesc = (TextView) findViewById(R.id.description_label);

		lblName.setText(name);
		lblCost.setText(cost);
		lblDesc.setText(description);
	}
}

File: kotlin/fix/XMLParsingTest/app/src/main/java/com/dstrube/xmlparsingtest/view/SingleMenuItemActivity.kt
package com.dstrube.xmlparsingtest.view

import android.widget.TextView
import android.content.Intent.getIntent
import android.content.Intent
import android.os.Bundle
import android.app.Activity
import com.dstrube.xmlparsingtest.R


class SingleMenuItemActivity : Activity() {

    // XML node keys
    private val KEY_NAME = "name"
    private val KEY_COST = "cost"
    private val KEY_DESC = "description"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.single_list_item)

        // getting intent data
        val `in` = intent //here's how to name a variable after a keyword :-p

        // Get XML values from previous intent
        val name = `in`.getStringExtra(KEY_NAME)
        val cost = `in`.getStringExtra(KEY_COST)
        val description = `in`.getStringExtra(KEY_DESC)

        // Displaying all values on the screen
        val lblName = findViewById(R.id.name_label) //as TextView
        val lblCost = findViewById(R.id.cost_label) //as TextView
        val lblDesc = findViewById(R.id.description_label) //as TextView

        lblName.text = name
        lblCost.text = cost
        lblDesc.text = description
    }
}